modules/dxp/apps/search-experiences/search-experiences-service/src/main/java/com/liferay/search/experiences/service/impl/SXPElementLocalServiceImpl.java

```diff
@@ -196,6 +197,10 @@ private void _validateExternalReferenceCode(
 			long companyId, String externalReferenceCode)
 		throws PortalException {
 
+		if (Validator.isNull(externalReferenceCode)) {
```

Comment:
I think we should not do this because of incorrect semantics: now it validates cleanly if ERC is null and we should not even allow that.

Reply:
Same here: I checked a couple of `_validateExternalReferenceCode` method in the codebase and found that they do this. But I also agree with you that it's not the best approach, though BChan may wants it.

Reply:
I would not do it like this. It allows null ERC and this way it would, theoretically, also allow there to be multiple elements with the same ERC: null watering the whole purpose of duplicate detection.

If ERC was null it should at least throw a proper exception.

Reply:
Ok, I can remove the null check. Will add back if BChan complaints.

Reply:
I think null check is good and should be there, eventually. It's just that it should not pass with null value. I don't see any reason to allow that.

Reply:
validator.isNull checks for empty too

Reply:
https://github.com/liferay/liferay-portal/blob/master/portal-kernel/src/com/liferay/portal/kernel/util/Validator.java#L925

Reply:
I'm not sure I follow. With that change and with ERC null, you would now return from _validateExternalReferenceCode() as if everything was fine. Instead an exception should be thrown if ERC null.

Reply:
Oh, sorry, I take my words back...

We actually seem to allow it. ERCs are not generated for user created elements. I don't understand why, but if this is intended then this null check is fine as is.

Reply:
This indeed seems to be the pattern. My bad.

So, all good to go.